mathlibtools/post-checkout

```diff
@@ -10,13 +10,13 @@ CHANGED_BRANCH=$3
 if [ "$CHANGED_BRANCH" -eq "1" ]; then
 	if /usr/bin/env python3 -c ""; then
 		echo "Trying to fetch cached olean"
-		leanproject get-cache
+		/usr/bin/env python3 -m leanproject get-cache
```

Comment:
(This may be the intention), but this will break in the "reverse" case -- where `leanproject` is on `PATH`, but `python3` is not the interpreter it is installed to.

Reply:
Correct - but in that situation the test for python3 is already nonsense.

Reply:
Concretely, this would break on nixos:
```
$ leanproject help
Usage: leanproject [OPTIONS] COMMAND [ARGS]...
Try 'leanproject -h' for help.

Error: No such command 'help'.
$ python3 -m leanproject
/run/current-system/sw/bin/python3: No module named leanproject
```

The `env python3` check also looks bogus to me.  Shouldn't it be `command -v leanproject >/dev/null`?  (I don't use the hooks though.)

Reply:
Ah, this line has a typo - it should have been `python3 -m mathlibtools.leanproject`.

Reply:
That's not the issue, none of the leanproject modules are in the PYTHONPATH even though `leanproject` is installed.
```
$ leanproject help
Usage: leanproject [OPTIONS] COMMAND [ARGS]...
Try 'leanproject -h' for help.

Error: No such command 'help'.
$ python3 -m mathlibtools.leanproject
/run/current-system/sw/bin/python3: Error while finding module specification for 'mathlibtools.leanproject' (ModuleNotFoundError: No module named 'mathlibtools')
```

Reply:
In that case, the issue is that `python3` is not referring to the same version of python as the one in the shebang of `leanproject` (all the more reason not to check for `python3`). What does `head $(which leanproject)` give?

Reply:
No, the `python3` is the same in both cases.
```
$ tail -n1 (which leanproject)
exec -a "$0" "/nix/store/21c77l6kirhlngw6gc0ky9r9rlmncll6-mathlib-tools-0.0.9b/bin/.leanproject-wrapped"  "$@"
$ head -n1 /nix/store/21c77l6kirhlngw6gc0ky9r9rlmncll6-mathlib-tools-0.0.9b/bin/.leanproject-wrapped
#!/nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8
$ readlink (which python3.8)
/nix/store/z65l1jqvxa58zzwwa3bvglb6asj4y8cv-python3-3.8.5/bin/python3.8
```

However `leanproject` changes the python path before it runs.

Reply:
Ok, I see what's happening. The default pip behave is to behave like both the nixos `buildPythonPackage` and `buildPythonApplication` at the same time.

You've presumably installed directly via the nixos method, and not via pip.